ui/src/components/admin/Model/ModelSingleHeader.js

```diff
@@ -100,20 +101,19 @@ const ModelSingleHeader = ({ modelName }) => {
   useEffect(() => {
     const refreshModelData = async () => await fetchModelData(modelName);
 
-    if (modelName && isPublishingModel(modelName) && (!publishState || !publishState.current)) {
+    if (modelName && isPublishingModel(modelName)) {
       publishState.current = 'publishing';
     }
 
     if (
       modelName &&
-      publishState.current === 'publishing' &&
+      (publishState.current === 'publishing' || !publishState.current) &&
       publishProgress.success.find(model => model.modelName === modelName)
     ) {
       publishState.current = 'published';
       refreshModelData();
-      return;
     }
-  }, [publishRunning, publishProgress, modelName, isPublishingModel, fetchModelData]);
```

Comment:
Removing unnecessary deps (added the linter ignore flag atop the file) since they were causing the useEffect to run unnecessarily.